### ISSUE TITLE
Device not connected or unavailable after every computer restart

### ISSUE THREAD

**Issue by DarkestFantasies** [obsproject/obs-studio#10769]
### Operating System Info

Windows 11

### OBS Studio Version

30.1.2

### OBS Studio Log URL

https://obsproject.com/logs/U9xfrwd98Idv7ohV

### OBS Studio Crash Log URL

https://obsproject.com/logs/OLaJBg2xHB3r8Sth

### Expected Behavior

My main monitor is selected by default

### Current Behavior

This setting forgets my main monitor

### Steps to Reproduce

1. Have 3 monitors on your system
2. Set your monitor in screen capture
3. Restart
...

### Anything else we should know?

![изображение](https://github.com/obsproject/obs-studio/assets/111584071/ed6a9c52-4f72-4647-9501-d006b342b41d)

**Comment by DarkestFantasies** [comment]
> Possible duplicate of #8181?
> 
> Please check this comment and provide the info requested: [#8181 (comment)](https://github.com/obsproject/obs-studio/issues/8181#issuecomment-1519167663)

no, it sometimes becomes "Device not connected or unavailable" again

**Comment by RytoEX** [comment]
Whether it happens "sometimes" or "always" is probably irrelevant. Do you have disabled display devices when it does happen? If so, then it's the same issue.

**Comment by DarkestFantasies** [comment]
![изображение](https://github.com/obsproject/obs-studio/assets/111584071/a7bf29a2-0c71-40fb-a3fc-7f1c7afe4682)


> Whether it happens "sometimes" or "always" is probably irrelevant. Do you have disabled display devices when it does happen? If so, then it's the same issue.

**Comment by DarkestFantasies** [comment]
> Whether it happens "sometimes" or "always" is probably irrelevant. Do you have disabled display devices when it does happen? If so, then it's the same issue.

seems like it never happens with my second monitor, i think it is because i am switching my primary monitor to my TV and back to my first monitor, and while i do it something going wrong with these settings

**Comment by RytoEX** [comment]
> > Whether it happens "sometimes" or "always" is probably irrelevant. Do you have disabled display devices when it does happen? If so, then it's the same issue.
> 
> seems like it never happens with my second monitor, i think it is because i am switching my primary monitor to my TV and back to my first monitor, and while i do it something going wrong with these settings

If the TV is turning off at any point in time while the PC is on, it is probably considered an "unavailable" or "disabled" display, and then Windows may change the order of its displays. TVs don't behave the same as PC displays.

**Comment by DarkestFantasies** [comment]
My TV doesn't actually turn off completely, it's in sleep mode.
![изображение](https://github.com/obsproject/obs-studio/assets/111584071/05bd8ca3-1e84-4f8c-9c0c-4e6831a04d2a)
and by default the signal doesn't go to the TV unless I turn it on, so this display is already "off" so to speak

**Comment by KasparsVanags** [comment]
Same issue but it doesn't happen every restart, only after leaving my PC off overnight.
Same setup - 2 monitors and a disabled TV.

Log from a failed detection:
https://obsproject.com/logs/LOXRGNG2T3mfBYDw

Log from a normal startup:
https://obsproject.com/logs/o2gh9Z3CmoV5WslE

Both displays are visible in the display selection list.

**Comment by DarkestFantasies** [comment]
> Same issue but it doesn't happen every restart, only after leaving my PC off overnight. Same setup - 2 monitors and a disabled TV.
> 
> Log from a failed detection: https://obsproject.com/logs/LOXRGNG2T3mfBYDw
> 
> Log from a normal startup: https://obsproject.com/logs/o2gh9Z3CmoV5WslE
> 
> Both displays are visible in the display selection list. When detection fails there's a selected "unconnected" display and the 2 monitors.

have you fixed ?

**Comment by CptLegshot** [comment]
> > Same issue but it doesn't happen every restart, only after leaving my PC off overnight. Same setup - 2 monitors and a disabled TV.
> > Log from a failed detection: https://obsproject.com/logs/LOXRGNG2T3mfBYDw
> > Log from a normal startup: https://obsproject.com/logs/o2gh9Z3CmoV5WslE
> > Both displays are visible in the display selection list. When detection fails there's a selected "unconnected" display and the 2 monitors.
> 
> have you fixed ?

Sort of but not completely, i disabled sleep mode on my TV and it happens very rarely now.
